pulser-core/pulser/__init__.py

```diff
@@ -14,9 +14,58 @@
 
 """A pulse-level composer for neutral-atom quantum devices."""
 
-from pulser._version import __version__
+from pulser._version import __version__ as __version__
+from pulser.waveforms import (
+    CompositeWaveform,
+    CustomWaveform,
+    ConstantWaveform,
+    RampWaveform,
+    BlackmanWaveform,
+    InterpolatedWaveform,
+    KaiserWaveform,
+)
 from pulser.pulse import Pulse
 from pulser.register import Register, Register3D
+from pulser.devices import AnalogDevice, DigitalAnalogDevice, MockDevice
 from pulser.sequence import Sequence
+from pulser.backend import (
+    EmulatorConfig,
+    NoiseModel,
+    QPUBackend,
+)
 
-from pulser.backend import QPUBackend  # isort: skip
+# Exposing relevant submodules
+from pulser import (
+    waveforms as waveforms,
+    channels as channels,
+    register as register,
```

Comment:
Should we consider register.special_layouts as relevant ? Since people should define their Register from a RegisterLayout for it to work on a real QPU, perhaps having `pulser.TriangularLatticeLayout` and `pulser.SquareLatticeLayout` would make sense ?

Reply:
I always hesitate on this one because, in the envisioned usage, the user should not be creating their own register layouts but rather taking choosing from the ones that are pre-calibrated. In practice, though, this is not what we observe (at least for now) so I do see value in moving the layout classes at least one level up (to the `register` module level).

As for the `pulser` namespace, I am making this sort of divide where I only expose  what I think "standard" users should be accessing. Perhaps we want to start encouraging the creating of registers from layouts, in which case it does make sense to move these all the way there?

Reply:
That's true, I figured this out after writing my question... I think it could make sense to move it one level up, just such that the user does not have to look to deep in the module to fetch it (makes sense to get them with pulser.register.AnyLayout)

I think history goes toward encouraging the creation of registers from layouts, but until we have a device where the user can define custom layouts, users should pick the layout in the calibrated_layouts of the device and there is no need to advertise about them too much (I fear it might just induce wrong layouts being used).

Reply:
Alright, let's be cautious and go only one level up for now